scripts/kconfig/kconfig.py

```diff
@@ -5,6 +5,35 @@
 
 from kconfiglib import Kconfig, Symbol, BOOL, STRING, TRISTATE, TRI_TO_STR
 
+# Warnings that won't be turned into errors (but that will still be printed),
+# identified by a substring of the warning. The warning texts from Kconfiglib
+# are guaranteed to not change.
+WARNING_WHITELIST = (
+    # Warning generated when a symbol with unsatisfied dependencies is being
+    # selected. These should be investigated, but whitelist them for now.
+    "unsatisfied direct dependencies",
+
+    # Assignments to certain undefined symbols. These are only defined for ARC
+    # and x86, respectively, but are set in some "shared" .conf files.
+    "undefined symbol ARC_INIT",  # Issue #7977
+    "undefined symbol SSE",       # Issue #7978
+
+    # Warning generated when a symbol is set more than once in a .config file.
+    # These should be investigated, but whitelist them for now.
+    "set more than once",
```

Comment:
Merging two different Kconfig fragments with overlapping and agreeable requirements should not produce a warning.

Reply:
If someone messes up and puts in the same assignment twice, it's cool to point it out.

There are some cases where it makes sense to have more than one assignment to the same variable though, when merging different combinations of `.config` files. Might even be okay to override assignments from the base configuration with a different value in some cases.

It's a tradeoff. If you think those redundant assignment warnings (when you give the same value both times) are more annoying than helpful, I could disable them though.

Reply:
I do. Please disable.